igvc_perception/src/multiclass_segmentation/train.py

```diff
@@ -0,0 +1,219 @@
+ap = argparse.ArgumentParser()
+
+ap.add_argument(
+    "-train_images",
+    "--train_images",
+    required=True,
+    help="path to train images .npy file",
+)
+ap.add_argument(
+    "-train_masks", "--train_masks", required=True, help="path to train masks .npy file"
+)
+ap.add_argument(
+    "-test_images", "--test_images", required=True, help="path to test images .npy file"
+)
+ap.add_argument(
+    "-test_masks", "--test_masks", required=True, help="path to test mask .npy file"
+)
+
+args = vars(ap.parse_args())
+
+train_images_path = args["train_images"]
+train_masks_path = args["train_masks"]
+test_images_path = args["test_images"]
+test_masks_path = args["test_masks"]
+
+# Set a seed for reproducibility
+SEED = 42
+utils.set_global_seed(SEED)
+utils.prepare_cudnn(deterministic=True)
+
+# Set up U-Net with pretrained EfficientNet backbone
+
+ENCODER = "efficientnet-b3"
+ENCODER_WEIGHTS = "imagenet"
+DEVICE = "cuda"
+
+ACTIVATION = None
+
+model = smp.Unet(
+    encoder_name=ENCODER,
+    encoder_weights=ENCODER_WEIGHTS,
+    classes=3,
+    activation=ACTIVATION,
+)
+
+# Get Torch loaders
+loaders = get_loaders(
+    images=np.load(train_images_path),
+    masks=np.load(train_masks_path),
+    image_arr_path=train_images_path,
+    mask_arr_path=train_masks_path,
+    random_state=420,
+    valid_size=0.1,
+    batch_size=3,
+    num_workers=2,
+)
+
+# Optimize for cross entropy using Adam
+criterion = {
+    "CE": CrossentropyND(),
+}
+
+learning_rate = 0.001
+encoder_learning_rate = 0.0005
+encoder_weight_decay = 0.00003
+optimizer_weight_decay = 0.0003
+optim_factor = 0.25
+optim_patience = 2
+
+optimizer = AdamW(
+    model.parameters(),
+    lr=0.001,
+    betas=(0.9, 0.999),
+    eps=1e-08,
+    weight_decay=0.01,
+    amsgrad=False,
+)
+
+scheduler = optim.lr_scheduler.ReduceLROnPlateau(
+    optimizer, factor=optim_factor, patience=optim_patience
+)
+
+num_epochs = 10
+device = utils.get_device()
+
+runner = SupervisedRunner(device=device, input_key="image", input_target_key="mask")
+
+# Use Catalyst callbacks for metric calculations during training
+callbacks = [
+    CriterionCallback(input_key="mask", prefix="loss", criterion_key="CE"),
+    MulticlassDiceMetricCallback(input_key="mask"),
+]
+
+# Train and print model training logs
+runner.train(
+    model=model,
+    criterion=criterion,
+    optimizer=optimizer,
+    scheduler=scheduler,
+    loaders=loaders,
+    callbacks=callbacks,
+    logdir="content/full_model2",
+    num_epochs=num_epochs,
+    main_metric="loss",
+    minimize_metric=True,
+    fp16=None,
+    monitoring_params=None,
+    verbose=True,
+)
+
+# Test model on test dataset
+test_data = SegmentationDataset(test_images_path, test_masks_path)
+infer_loader = DataLoader(test_data, batch_size=12, shuffle=False, num_workers=4)
+
+# Get model predictions on test dataset
+predictions = np.vstack(
+    list(
+        map(
+            lambda x: x["logits"].cpu().numpy(),
```

Comment:
This lambda is also broken:
```
Traceback (most recent call last):
  File "train.py", line 162, in <module>
    loader=infer_loader, resume=f"content/full_model2/checkpoints/best.pth", model=model
  File "train.py", line 160, in <lambda>
    lambda x: x["logits"].cpu().numpy(),
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

Reply:
Found the issue. This is a Catalyst 20.4.2 feature; I just updated requirements.txt